### ISSUE TITLE
questions about whole tied symbols

### ISSUE THREAD

**Issue by kkp15** [espeak-ng/espeak-ng#1065]
Thank you for this amazing repository!
I have a question regarding the IPA symbols.
I'd like to know the whole set of tied IPA symbols when using espeak.
Is there any way I can get this set?

**Comment by jaacoppi** [comment]
I don't know what a tied IPA symbol is, but you can see the phonemes in phsource/phonemes. Does that help?

**Comment by kkp15** [comment]
Yes! Thank you very much. But is there anyway I can get the whole ipa phoneme set that includes every ipa symbol for entire language in espeak?

**Comment by jaacoppi** [comment]
In short, there's no quick way. See the docs in docs/phontab.md

Check the phsource/ directory. There's ph_english for English, ph_finnish for Finnish and so on.
To see what file matches what language, you'll need to check the phonemetable entry in phsource/phonemes.
For example:
phonemetable fi base1
include ph_finnish

This means that for phoneme setting "fi", the file is ph_finnish. For most languages the phonemetable is just the language code. So, phonemetable "fi" for language "fi" is in file ph_finnish.
Some languages share a phoneme table. See files in espeak-ng-data/lang/*/*, keyword "phonemes". 

For example:
espeak-ng-data/lang/zls/bs:phonemes hr
So, language bs uses phonemetable hr. Then cross reference in phsource/phonemes to see which file ha phonemetable hr:
phonemetable hr sk
include ph_croatian
So, phonemes for languge bs are in file ph_croatian. There's also the parent table (base1 and sk in the examples). Read the docs for more info.






-- 
Juho Hiltunen

**Comment by valdisvi** [comment]
You can get all IPA symbols by simply searching all of them in `phsource` folder of the project:

```
grep -Er "^\s*ipa" # see in which files they are defined
grep -Er "^\s*ipa"|awk -F "//" '{print $1}'|awk '{print $NF}'|sort -u # print only phonemes
```
But, not all phoneme definitions have this entry #769 and then phoneme name will be printed out instead. Then you have to search all phoneme definitions as well:

```
grep -Er "^phoneme|^\s*ipa"|awk -F "//" '{print $1}'|awk '{print $NF}'|sort -u
```

**Comment by kkp15** [comment]
Thank you so much @valdisvi ! 
This is exactly what I needed :)
Seems like there are some issues left still.
For example, if I run the command you suggested, I see some weird outputs which are not phonemes (sometimes words and numbers)
Some symbols that appear first include:
!
#
#X1
#X2
%
%%
&
&#
&/
&:
&U~
&i
&y
&~
'
'!
''
*
**
*;
+
+2
-
0
0#
02
0`
0~
0~`
1
11
2
21
214
22
23
3
3-
32
33
35
3:
4
43
44
5
51
53
55
6
7
8
8:
:
:0
:1
;
<h>

In addition, I wonder why some of the symbols are not converted to actual symbols but in a utf-8 form
e.g. "oU+0303ɪU+0303"

Looking forward to your answer!
